src/NSwag.CodeGeneration.CSharp/Templates/Client.Class.liquid

```diff
@@ -405,6 +405,14 @@
         public string Text { get; }
     }
 
+    private bool StatusInRange(int value, string match)
+    {
+        var regexBody = match.Replace("x", "\\d", System.StringComparison.OrdinalIgnoreCase);
```

Comment:
Building a regex from external input should usually be avoided. Consider parsing the first digit using a pattern like `@"^(?<class>\d)xx$`. This would also allow statically holding a pre-compiled regex instead of creating a new one on every request, thus avoiding multiple compilations. You could also consider not using a regex altogether and instead parsing completely old fashioned (match.length == 3) && (match[1] == ...).

Reply:
Please see update. I made the regex part of the generated compile time code.